### ISSUE TITLE
Build failure: lokinet

### ISSUE THREAD

**Issue by alciono** [NixOS/nixpkgs#285949]
### Steps To Reproduce

1. insert lokinet in configuration.nix file
2. sudo nix-channel --update
3. sudo nixos-rebuild switch --upgrade  --show-trace

### Build log

```
/build/source/llarp/util/thread/queue_manager.cpp: In member function 'void llarp::thread::QueueManager::abortPushIndexReservation(uint32_t, uint32_t)':
/build/source/llarp/util/thread/queue_manager.cpp:491:33: error: 'popIndex' was not declared in this scope
  491 |       uint32_t loadedPopIndex = popIndex().load(std::memory_order_relaxed);
      |                                 ^~~~~~~~
/build/source/llarp/util/thread/queue_manager.cpp: In member function 'size_t llarp::thread::QueueManager::size() const':
/build/source/llarp/util/thread/queue_manager.cpp:509:56: error: 'pushIndex' was not declared in this scope
  509 |       uint32_t combinedPushIndex = discardDisabledFlag(pushIndex());
      |                                                        ^~~~~~~~~
/build/source/llarp/util/thread/queue_manager.cpp:510:35: error: 'popIndex' was not declared in this scope
  510 |       uint32_t combinedPopIndex = popIndex();
      |                                   ^~~~~~~~
/build/source/llarp/util/thread/queue_manager.cpp: In member function 'bool llarp::thread::QueueManager::enabled() const':
/build/source/llarp/util/thread/queue_manager.cpp:543:33: error: 'pushIndex' was not declared in this scope
  543 |       return !isDisabledFlagSet(pushIndex().load());
      |                                 ^~~~~~~~~
make[2]: *** [llarp/CMakeFiles/lokinet-util.dir/build.make:202: llarp/CMakeFiles/lokinet-util.dir/util/thread/queue_manager.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 37%] Building CXX object external/oxen-mq/CMakeFiles/oxenmq.dir/oxenmq/proxy.cpp.o
[ 37%] Building CXX object external/oxen-mq/CMakeFiles/oxenmq.dir/oxenmq/worker.cpp.o
make[1]: *** [CMakeFiles/Makefile2:567: llarp/CMakeFiles/lokinet-util.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 38%] Linking CXX shared library liboxenmq.so
[ 38%] Built target oxenmq
make: *** [Makefile:136: all] Error 2
error: builder for '/nix/store/aj291y7nzjkr6gwcscl23ayfg1vynvv7-lokinet-0.9.11.drv' failed with exit code 2

```

### Metadata

 nix-shell -p nix-info --run "nix-info -m"                                                                                    255 ↵ ──(Sat,Feb03)─┘
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.74, NixOS, 24.05 (Uakari), 24.05pre578631.b8b232ae7b8b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos, nixos-unstable, nixpkgs, nixpkgs-unstable"`
 - channels(k): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by wyndon** [comment]
https://github.com/NixOS/nixpkgs/pull/281069

**Comment by FliegendeWurst** [comment]
Fixed in #287709
